CONTRIBUTING.md

```diff
@@ -0,0 +1,172 @@
+
+#### Reference
+
+If the section is for a data structure, it should include a subsection containing skeleton reference code.
+The reference code should include only what is necessary to use the data structure and perhaps usage examples in a main function if its use is not obvious enough.
+
+#### Applications
+
+A list of uses for the algorithm or data structure.
+For example, one application of a maximum flow algorithm is 'finding a minimum set of edges to disconnect a source and sink'.
+
+#### Sample Problem
+
+A problem whose solution utilizes the algorithm or data structure the section describes.
+Each problem should include a description that _clearly_ describes the problem. Following the problem description should be as follows:
+
+##### Input Format
+
+A precise description of the input that will be supplied to the code that will solve the problem described.
+
+##### Sample Input
+
+An example input that fully demonstrates and conforms to the format described in 'Input Format'.
+Use the `\acmlisting` command to insert your sample input.
+See the 'Code Listings' 
+
+##### Output Format
+
+A precise description of the output that should be produced from the code that will solve the problem described.
+
+##### Sample Output
+
+An example input that conforms to the format described in 'Output Format'.
+Use the `\acmlisting` command to insert your sample output.
+See the 'Code Listings' section under 'Format' for more information.
+
+##### Sample Solution
+
+A solution for the problem described.
+All solutions should be written in C++ __before__ providing solutions in other languages.
+
+##### Lessons Learned
+
+An unordered list of at least one non-trivial observation you had while writing the solution to your sample problem.
+
+***
+
+### Figures and Images
+
+If adding a visual to a section would greatly aid in the understanding of the topic, doing so is highly encouraged.
+Submissions should preferably be in the Encapsulated PostScript format.
+This provides very clear visuals, even when scaled up or down.
+Do _not_ simply pull images from the Internet.
+Doing so may violate the Hack Pack licensing terms.
+Any submission found to be doing so will be rejected until the appropriate changes are made.
+
+### Indexing
+
+Indexing entries is very helpful for finding relevant entries in the Hack Pack.
+Please do not sprinkle index entries _too_ liberally throughout, as this will cause the size of the index to explode and make it harder to locate information.
+An example of a good set of index entries can be found in the section on the Sieve of Eratosthenes.
+Entries include:
+
+- sieve
+- primes
+- Eratosthenes
+
+### Citations
+
+Information pulled from other sources needs to be properly cited.
+Any information included in submissions that is not properly cited will cause the entire section to be rejected from consideration for inclusion.
+References should use BibTeX where applicable.
+
+## Format
+
+### General Rules to Type By
```

Comment:
We should include some reasons for these rules. for example:

+ Only one line per sentence.  This makes it easier to review changes to documentation.
+ Use tabs for indentation.  This allows editors to choose their preferred tab width.
+ Do not use LaTeX comments.  Anything important enough to put in a comment should be put in the contents.  Exceptions would include the `style.tex` file.

I also think that we should address when indentation should be used.